### PR TITLE
Add telemetry knobs for metrics

### PR DESCRIPTION
**Pull request by rgrandl**
Right now the user doesn't have any way to control how metrics are being manipulated by the framework. However, the framework takes certain decisions that can significantly impact the cloud bill:
* for example, the framework generates application metrics (e.g., method count, error, latency, bytes sent/received), and also metrics for our system components (e.g., nanny, babysitter);
* the framework exports the metrics to Google Cloud every 15 seconds.

This PR enables the user to control (1) whether they want autogenerated metrics and (2) how often to export the metrics to Google Cloud.

To do that, the user has to add a new config under the gke section like:
```
[gke]
telemetry.metrics = {export_interval = "1h", auto_generate_metrics =
true}
```

If not specified by the user, we'll export metrics every 30s, and we'll not export any autogenerated metrics.

In future PRs, we'll add config knobs for logging and traces as well.

### CODE REVIEW

File: internal/babysitter/babysitter.go [hunk 1]
```diff
@@ -158,8 +158,7 @@ func (b *Babysitter) SelfAddr() string { return b.selfAddr }
 // exportMetrics periodically exports metrics.
 func (b *Babysitter) exportMetrics() {
 	// Time interval at which metrics are exported.
-	const metricExportInterval = 15 * time.Second
-	ticker := time.NewTicker(metricExportInterval)
+	ticker := time.NewTicker(b.cfg.Telemetry.Metrics.ExportInterval.AsDuration())
```
Comment (spetrovic77) [inline internal/babysitter/babysitter.go:161]: What happens if the `ExportInterval` is zero? Can we explicitly check for this?

File: internal/gke/metrics.go [hunk 2]
```diff
@@ -114,6 +116,25 @@ func (m *metricExporter) Close() error {
 	return m.client.Close()
 }
 
+// filterMetrics returns a set of filtered metrics.
+func filterMetrics(snaps []*metrics.MetricSnapshot, exportGeneratedMetrics bool) []*metrics.MetricSnapshot {
```
Comment (spetrovic77) [inline internal/gke/metrics.go:120]: We need to let the metric used during rollouts go through. This is annoying, we can chat about it in office.

**Review by spetrovic77 — COMMENTED** [review]
Looks great.

File: internal/babysitter/babysitter.go (hunk 1 above)
Comment (rgrandl) [reply internal/babysitter/babysitter.go:161]: Good point. I did the validation in `internal/tool/deploy` when we build the options; we enforce `export_interval` to be at least 10 seconds, wdyt?

File: internal/gke/metrics.go (hunk 2 above)
Comment (rgrandl) [reply internal/gke/metrics.go:120]: Yeah, I already did that change yesterday: https://github.com/ServiceWeaver/weaver/blob/main/http.go#L96. For the httpRequestCount, we don't label it as generated.

**Review by rgrandl — COMMENTED** [review]
Thanks, Srdjan

File: internal/babysitter/babysitter.go (hunk 1 above)
Comment (spetrovic77) [reply internal/babysitter/babysitter.go:161]: SGTM

File: internal/gke/metrics.go (hunk 2 above)
Comment (spetrovic77) [reply internal/gke/metrics.go:120]: ACK!

**Review by spetrovic77 — APPROVED** [review]

**Pull request merged**